### ISSUE TITLE
Commits in the last day seem to have reset training progress

### ISSUE THREAD

**Issue by leijurv** [deepfakes/faceswap#159]
Loss has jumped back up to the original values, and are descending no faster than when I originally started training a few days ago.

Is this an intended change? Will future commits require a full retrain?

**Comment by Clorr** [comment]
Hmm, it shouldn't and also the loss cannot go back, unless the loss function is modified which is not the case. Assuming you have the same config / arguments, I only see 3 possible behaviors:
- Paths have changed / Filenames have changed: A message will indicate files have not been found
- Model has changed / Files are corrupted / An updated lib makes files unreadable: A message will indicate files can't be reloaded (In this case, be careful, the existing files will be overwritten by a new model save)
- Otherwise the model will be reloaded as is with it's current progress

Note that the message may not be much distinguishible from other outputs

**Comment by leijurv** [comment]
I believe this could be the cause:
https://github.com/deepfakes/faceswap/commit/b3ae6130ed4714852211a6e8cb56de66450da83c#diff-298a939aa6a53af15be411af8f08ac8fR43

The training data has changed. Now it has random transforms

**Comment by leijurv** [comment]
`https://github.com/deepfakes/faceswap/commit/b3ae6130ed4714852211a6e8cb56de66450da83c#diff-298a939aa6a53af15be411af8f08ac8fR43`

**Comment by leijurv** [comment]
I understand the purpose of random mirrors and flips, but what is the purpose of randomly transforming (rotating, shifting, zooming) the faces after I spent hours aligning them the exact same way with extract?

**Comment by Clorr** [comment]
Random transform args were previously in training_data. They were not added. Also it wouldn't change much on output loss. AFAIK this part is to avoid having always the same image as source, it increases diversity.

Also note the values are very small, and aligned faces , even very similar ones, will have bigger differences

**Comment by leijurv** [comment]
I don't see this randomness in the previous version https://github.com/deepfakes/faceswap/blob/b34d7d1a4956ae4f3eaf863f071e3f70e301a097/lib/ModelAE.py#L46

**Comment by leijurv** [comment]
With regards to the earlier question: I stopped training, git pulled, and restarted training and the loss increased dramatically (from 0.016 to 0.042). No error was printed, and no paths were changed.

```
loaded model weights
Loading Trainer from Model_Original plugin...
Starting. Press "Enter" to stop training and save model
saved model weights loss_A: 0.04229, loss_B: 0.04221
```

**Comment by Clorr** [comment]
Random_transform_args where in lib/training_data.py line 10.

Can you tell me which was the previous version you were using? Because one thing that changed lately is the face coverage that changed from 160 to 220. So maybe that is a possibility. Can you try setting it to 160 again and see if it changes loss?

**Comment by leijurv** [comment]
I originally trained on b34d7d1a4956ae4f3eaf863f071e3f70e301a097. All commits on Feb 7 and Feb 8 are the new ones.

**Comment by IUsedToBeAPygmy** [comment]
I've just synced and looking at the preview window it looks like the faces are much more zoomed in now - to the point where some faces where the mouth is open actually have the mouth go 'out of view'.

Could it be that an extra zoom factor was accidentally added?

**Comment by Clorr** [comment]
Hmm, not willingly for sure... I should check where this can come from, however I won't be available until Monday

**Comment by qzmenko** [comment]
> I've just synced and looking at the preview window it looks like the faces are much more zoomed in now - to the point where some faces where the mouth is open actually have the mouth go almost go out of frame. Could it be that an extra zoom factor was accidentally added somehow?
I confirm that. Images in preview window looks more zoomed.

**Comment by IUsedToBeAPygmy** [comment]
I'm at work now so can't check, but maybe someone can go back a few versions to find out in which changelist this first happened?

**Comment by IUsedToBeAPygmy** [comment]
I think I've found it:

In Clorr's changelist 
https://github.com/deepfakes/faceswap/commit/b3ae6130ed4714852211a6e8cb56de66450da83c#diff-cd6853ac48b3d61ae96f466bcc2c393eR41

The face coverage has changed from 220 to 160:

> coverage = 220 # Coverage of the face for training. Larger value will cover more features. @shaoanlu recommends 220. Original is 160

to

> TrainingDataGenerator(self.random_transform_args, 160)

**Comment by Clorr** [comment]
@IUsedToBeAPygmy the coverage is a feature from the very original repo. I introduced a var in order to be able to have 2 different values because faceswap-GAN repo does not have the same value.

AFAIK the face coverage is the part that is warped and if you did apply the warping to the whole image, you would loose part of the image and then have problem with your training

Also having it bigger wouldn't give more details to the network as the network has an input that is 64x64

**Comment by IUsedToBeAPygmy** [comment]
@Clorr - seriously, check the code!

The coverage factor determines what "area" of the 256x256 image is assumed to contain the real face data to train on.


First a value of 220 was used for the coverage factor. Meaning, 220/256th of the image was used to train on.

Then the coverage factor was (re)set to 160 in your changelist, which takes only 160/256th of the image (taken from the center) as "relevant face data".
This caused the zoom in the preview window, as the area we're training on is smaller thus will be blown up more in the preview.

If you didn't *intentionally* change the coverage factor from 220 back to 160, I'd strongly suggest putting it at 220 again.

**Comment by Clorr** [comment]
No I did it intentionally following this issue: #96

If you want to add a CLI argument to give this var the value you want, feel free.
